### ISSUE TITLE
`Loader`'s `@loaded` always compares by identity

### ISSUE THREAD

**Issue by FunkyloverOne** [djezzzl/n1_loader#32]
Hey, this line here is problematic:
https://github.com/djezzzl/n1_loader/blob/7756cf8f9506eb6307a3788928119f55c2832921/lib/n1_loader/core/loader.rb#L111

Why isn't it configurable? :sweat_smile: What if I actually want it to compare normally? Here's the use case:
- the loader receives an `ActiveRecord::Relation` as its `elements`, 
- then in the `#perform` it actually loads those records, 
- and then calls the `#fulfill` for each separate record, 
- and then when the _different_ AR object (but **same** record)  is passed to the `#for` method - there's no such key :grimacing:

For now as a workaround I'm using the following hack:
```ruby
def loaded
  super.tap { @loaded = {}.merge!(@loaded) if @loaded.compare_by_identity? }
end
```

**Comment by djezzzl** [comment]
Hi @FunkyloverOne,

This is configurable through [`cache_key`](https://github.com/djezzzl/n1_loader/blob/master/examples/arguments_support.rb#L24)  if I understood your problem correctly. 

If this is what you expected, could you please help me to improve the documentation?

**Comment by FunkyloverOne** [comment]
Hey @djezzzl, no my problem is different. 
The `cache_key` is used to compare **loaders** between each other.
And the `loaded` - is a hash, where the keys are `elements` and the values are their loaded data.

**Comment by djezzzl** [comment]
Hi @FunkyloverOne,

Thank you for the issue, and I'm sorry it took me long to address it.

It was done here: https://github.com/djezzzl/n1_loader/pull/41.

And was released as 1.7.0. 

Please check it out and let me know if it helps.

Thank you, and have a great day!
